### PR TITLE
Fix crash when taking over some games on 5.0.0+

### PR DESCRIPTION
**Pull request by SciresM**
This fixes a system crash in AM when games with VideoCaptureMode = automatic are launched on 5.0.0+.

### CODE REVIEW

File: source/main.c [hunk 1]
```diff
@@ -14,6 +14,10 @@ static u64  g_nroSize = 0;
 static NroHeader g_nroHeader;
 static bool g_isApplication = 0;
 
+
```
Comment (yellows8) [inline source/main.c]: extra newline

File: source/main.c [hunk 2]
```diff
@@ -73,7 +77,11 @@ void setupHbHeap(void)
         size = (mem_available - mem_used - 0x200000) & ~0x1FFFFF;
     if (size==0)
         size = 0x2000000*16;
-
+    
+    if (g_isAutomaticGameplayRecording) {
```
Comment (yellows8) [inline source/main.c]: Should check size here.

File: source/main.c [hunk 3]
```diff
@@ -73,7 +76,11 @@ void setupHbHeap(void)
         size = (mem_available - mem_used - 0x200000) & ~0x1FFFFF;
     if (size==0)
         size = 0x2000000*16;
-
+
```
Comment (fincs) [inline source/main.c]: Trailing whitespace

File: source/main.c [hunk 4]
```diff
@@ -73,7 +76,11 @@ void setupHbHeap(void)
         size = (mem_available - mem_used - 0x200000) & ~0x1FFFFF;
     if (size==0)
         size = 0x2000000*16;
-
+    
+    if (size > 0x6000000 && g_isAutomaticGameplayRecording) {
+        size -= 0x6000000;
+    }
+
```
Comment (fincs) [inline source/main.c]: Trailing whitespace

File: source/main.c [hunk 5]
```diff
@@ -136,6 +143,29 @@ void getIsApplication(void) {
     if (R_SUCCEEDED(rc) && cur_pid == app_pid) g_isApplication = 1;
 }
 
+//Gets the control.nacp for the current title id, and then sets g_isAutomaticGameplayRecording if less memory should be allocated.
+void getIsAutomaticGameplayRecording(void) {
+    if (kernelAbove500() && g_isApplication) {
+        Result rc=0;
+        u64 cur_tid=0;
+
```
Comment (fincs) [inline source/main.c]: Trailing whitespace

File: source/main.c [hunk 6]
```diff
@@ -136,6 +143,29 @@ void getIsApplication(void) {
...
 
+//Gets the control.nacp for the current title id, and then sets g_isAutomaticGameplayRecording if less memory should be allocated.
+void getIsAutomaticGameplayRecording(void) {
+    if (kernelAbove500() && g_isApplication) {
+        Result rc=0;
+        u64 cur_tid=0;
+        
+        rc = svcGetInfo(&cur_tid, 18, CUR_PROCESS_HANDLE, 0);
+        if (R_FAILED(rc)) return;
+
```
Comment (fincs) [inline source/main.c]: Trailing whitespace

File: source/main.c [hunk 7]
```diff
@@ -136,6 +143,29 @@ void getIsApplication(void) {
...
+void getIsAutomaticGameplayRecording(void) {
+    if (kernelAbove500() && g_isApplication) {
+        Result rc=0;
+        u64 cur_tid=0;
+        
+        rc = svcGetInfo(&cur_tid, 18, CUR_PROCESS_HANDLE, 0);
+        if (R_FAILED(rc)) return;
+        
+        g_isAutomaticGameplayRecording = 0;
+
```
Comment (fincs) [inline source/main.c]: Trailing whitespace

File: source/main.c [hunk 8]
```diff
@@ -136,6 +143,29 @@ void getIsApplication(void) {
...
+        Result rc=0;
+        u64 cur_tid=0;
+        
+        rc = svcGetInfo(&cur_tid, 18, CUR_PROCESS_HANDLE, 0);
+        if (R_FAILED(rc)) return;
+        
+        g_isAutomaticGameplayRecording = 0;
+        
+        rc = nsInitialize();
+
```
Comment (fincs) [inline source/main.c]: Trailing whitespace

File: source/main.c [hunk 9]
```diff
@@ -136,6 +143,29 @@ void getIsApplication(void) {
...
+        g_isAutomaticGameplayRecording = 0;
+        
+        rc = nsInitialize();
+        
+        if (R_SUCCEEDED(rc)) {
+            size_t dummy;
+            rc = nsGetApplicationControlData(0x1, cur_tid, &g_applicationControlData, sizeof(g_applicationControlData), &dummy);
+            nsExit();
+        }
+
```
Comment (fincs) [inline source/main.c]: Lots of trailing whitespace

**Pull request merged**